client/src/common.rs

```diff
@@ -213,6 +213,66 @@ pub mod block_hash {
     }
 }
 
+pub mod block_parameter_type {
+    use super::*;
+
+    const ARG_NAME: &str = "parameter-type";
+    const ARG_NAME_SHORT: &str = "t";
+    const ARG_HELP: &str = "Flag toggle to query by either hash or height";
+
+    pub(crate) fn arg(order: usize) -> Arg<'static, 'static> {
+        Arg::with_name(ARG_NAME)
+            .short(ARG_NAME_SHORT)
+            .required(false)
+            .help(ARG_HELP)
+            .display_order(order)
+    }
+
+    pub(crate) fn get(matches: &ArgMatches) -> bool {
+        matches.is_present(ARG_NAME)
+    }
+}
+
+pub mod block_parameter {
+    use casper_node::rpcs::chain::BlockParameters;
+
+    use super::*;
+
+    const ARG_NAME: &str = "block-parameter";
+    const ARG_SHORT: &str = "b";
+    const ARG_VALUE_NAME: &str = "HEX_STRING";
```

Comment:
What is the reason that we are overloading a single flag to take either the hex or an integer? e.g. `--block-height=N` and `--block-hash=deadbeef...`

Reply:
I just feel it's simpler since there's not really any ambiguity about which is intended when the user enters it on the command line.

But maybe you're right and the proper thing to do is to have separate args.  Let's see what feedback we get from users.  If it's confusing we can change to your suggested approach.